src/sentry/static/sentry/app/views/dashboardsV2/widgetCard.tsx

```diff
@@ -69,6 +70,193 @@ class WidgetCard extends React.Component<Props> {
+              return (
+                <React.Fragment>
+                  {typeof renderErrorMessage === 'function'
+                    ? renderErrorMessage(errorMessage)
+                    : null}
+                  <ChartContainer>
+                    <HeaderTitleLegend>{widget.title}</HeaderTitleLegend>
+                    <WidgetCardVisuals
+                      timeseriesResults={timeseriesResults}
+                      tableResults={tableResults}
+                      errorMessage={errorMessage}
+                      loading={loading}
+                      location={location}
+                      widget={{
+                        ...widget,
+                        title: '',
+                      }}
+                      selection={selection}
+                      router={router}
+                    />
+                    {this.renderToolbar()}
+                  </ChartContainer>
+                </React.Fragment>
+              );
+            }}
+          </WidgetQueries>
+        </StyledPanel>
+      </ErrorBoundary>
+    );
+  }
+}
+
+export default withApi(
+  withOrganization(withGlobalSelection(ReactRouter.withRouter(WidgetCard)))
+);
+
+const ErrorCard = styled(Placeholder)`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: ${p => p.theme.alert.error.backgroundLight};
+  border: 1px solid ${p => p.theme.alert.error.border};
+  color: ${p => p.theme.alert.error.textLight};
+  border-radius: ${p => p.theme.borderRadius};
+  margin-bottom: ${space(2)};
+`;
+
+const StyledPanel = styled(Panel, {
+  shouldForwardProp: prop => prop !== 'isDragging',
+})<{
+  isDragging: boolean;
+}>`
+  margin: 0;
+  visibility: ${p => (p.isDragging ? 'hidden' : 'visible')};
+`;
+
+const ToolbarPanel = styled('div')`
+  position: absolute;
+  top: 0;
+  left: 0;
+  z-index: 1;
+
+  width: 100%;
+  height: 100%;
+
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  background-color: rgba(255, 255, 255, 0.5);
+`;
+
+const IconContainer = styled('div')`
+  display: flex;
+
+  > * + * {
+    margin-left: 50px;
+  }
+`;
+
+const IconClick = styled('div')`
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+const StyledIconGrabbable = styled(IconGrabbable)`
+  &:hover {
+    cursor: grab;
+  }
+`;
+
+type WidgetCardVisualsProps = Pick<ReactRouter.WithRouterProps, 'router'> &
+  Pick<
+    WidgetQueries['state'],
+    'timeseriesResults' | 'tableResults' | 'errorMessage' | 'loading'
+  > & {
+    location: Location;
+    widget: Widget;
+    selection: GlobalSelection;
+  };
+class WidgetCardVisuals extends React.Component<WidgetCardVisualsProps> {
+  shouldComponentUpdate(nextProps: WidgetCardVisualsProps): boolean {
+    // Widget title changes should not update the WidgetCardVisuals component tree
+    const currentProps = {
+      ...this.props,
+      widget: {
+        ...this.props.widget,
+        title: '',
+      },
+    };
```

Comment:
Would this be simpler as a `cloneDeep` operation? Something like

```ts
const currentProps = cloneDeep(this.props);
currentProps.widget.title = '';
```

Reply:
That might be more costly than a simple shallowclone. I don't intend to overwrite anything in the props.